cobra/manipulation/delete.py

```diff
@@ -15,22 +16,25 @@ def prune_unused_metabolites(cobra_model):
     Parameters
     ----------
     cobra_model: cobra.Model
-        the model to remove unused metabolites from
+    the model to remove unused metabolites from
 
     Returns
     -------
+    model
+    input model with unused metabolites removed
     list
-        list of metabolites that were removed
+    list of metabolites that were removed
     """
     inactive_metabolites = []
     active_metabolites = []
-    for the_metabolite in cobra_model.metabolites:
+    output_model = cobra_model.copy()
+    for the_metabolite in output_model.metabolites:
```

Comment:
Since the active metabolites are not actually used for anything the whole loop can be made into a list comprehension.
```python
isolated = [m for m in output_model.metabolites if len(m.reactions) == 0]
```